pyidr/study_parser.py

```diff
@@ -308,35 +325,109 @@ def check(obj):
+                        help="Fail if unknown keys are detected")
+    parser.add_argument("--inspect", action="store_true",
+                        help="Inspect the internals of the study directory")
     parser.add_argument("--report", action="store_true",
                         help="Create a report of the generated objects")
     parser.add_argument("--check", action="store_true",
                         help="Check against IDR")
+    parser.add_argument("--format", default="text", choices=("text", "json"),
+                        help="Format for the report")
     args = parser.parse_args(argv)
 
-    try:
-        for s in args.studyfile:
-            p = StudyParser(s)
-            objects = [Object(x) for x in p.components]
-            if args.report:
-                for o in objects:
-                    log.info("Generating annotations for %s" % o.name)
-                    print "description:\n%s\n" % o.description
-                    print "map:"
-                    print "\n".join(["%s\t%s" % (v[0], v[1]) for v in o.map])
-
-            if args.check:
-                for o in objects:
-                    log.info("Check annotations for %s" % o.name)
-                    check(o)
-    except Exception:
-        traceback.print_exc()
-        sys.exit(1)
+    if args.format == "json":
+        Printer = JsonPrinter
+    elif args.format == "text":
+        Printer = TextPrinter
+    else:
+        raise Exception("unknown:" + args.format)
+
+    for s in args.studyfile:
+        p = StudyParser(s)
+        unknown = []
+        for idx, line in enumerate(p._study_lines_used):
+            if not line:
+                line = p._study_lines[idx].strip()
+                if line and \
+                        not line.startswith('#') and \
+                        not line.startswith('"'):
+                    key = line.split("\t")[0]
+                    if args.strict:
+                        unknown.append(key)
+                    else:
+                        log.warn("Unknown key: %s", key)
+        if unknown:
+            print "Found %s unknown keys:" % len(unknown)
+            raise Exception("\n".join(unknown))
+        printer = Printer(p)
+        objects = [Object(x) for x in p.components]
+        if args.inspect:
+            basedir = os.path.dirname(s)
+            log.info("Inspect the internals of %s" % basedir)
+            for o in objects:
+                path = "%s/%s/*" % (basedir, o.name.split("/")[-1])
+                o.files = glob.glob(path)
+
+        if args.report:
+            for o in objects:
+                log.info("Generating annotations for %s" % o.name)
+                printer.consume(o)
+
+        if args.check:
+            for o in objects:
+                log.info("Check annotations for %s" % o.name)
+                check(o)
+        printer.finish()
+    return p
 
 
 if __name__ == "__main__":
-    main(sys.argv[1:])
+    parser = main(sys.argv[1:])
```

Comment:
What the expected scope of returning the parser? How should it behave when multiple study files are supplied?